comment/views/comments.py

```diff
@@ -45,22 +58,48 @@ def form_valid(self, form):
         model_id = self.request.POST.get('model_id')
         model_object = get_model_obj(app_name, model_name, model_id)
         parent_id = self.request.POST.get('parent_id')
-        parent_comment = None
-        if parent_id:
-            parent_qs = Comment.objects.filter(id=parent_id)
-            if parent_qs.exists():
-                parent_comment = parent_qs.first()
+        parent_comment = Comment.objects.get_parent_comment(parent_id)
+        if self.request.user.is_authenticated:
+            user = self.request.user
+        else:
+            user = None
+
         comment_content = form.cleaned_data['content']
-        self.comment = Comment.objects.create(
+        email = form.cleaned_data['email']
+        time_posted = timezone.now()
+        comment = Comment(
             content_object=model_object,
             content=comment_content,
-            user=self.request.user,
+            user=user,
             parent=parent_comment,
-        )
+            email=email,
+            posted=time_posted
+            )
+        dict_comment = {
+            'user': user,
+            'content': comment_content,
+            'email': email,
+            'posted': str(time_posted),
+            'app_name': app_name,
+            'model_name': model_name,
+            'model_id': model_id,
+            'parent': parent_id
+        }
+        if settings.COMMENT_ALLOW_ANONYMOUS and self.request.user.is_anonymous:
+            # send response, please verify your email to post this comment.
+            site = get_current_site(self.request)
+            key = signing.dumps(dict_comment, compress=True)
+            send_email_confirmation_request(comment, email, key, site)
+            return JsonResponse({
+                'msg': _("""We've have sent a verification link to your email. The comment will be displayed\
+                    after your verification.""")
+            })
+        comment.save()
+        self.comment = comment
         return self.render_to_response(self.get_context_data())
 
 
-class UpdateComment(BaseCommentView):
+class UpdateComment(BaseCommentMixin, AnonymousPermissionMixin, BaseCommentView):
```

Comment:
`AnonymousPermissionMixin` is useless here. dispatch method is already performing the check against comment owner and anonymous comment will never have owner.

Reply:
Yes you are right, I didn't see the dispatch method here. I will remove the mixin.